explorer existing integration test is failing

```
  OPS: Operations Per Second, computed as 1 / Mean
================================================================================================= short test summary info ==================================================================================================
FAILED integration_tests/test_page_loads.py::test_general_dataset_redirect - AssertionError: assert '/products/ls...-73f57de0574d' == 'http://local...-73f57de0574d'
FAILED integration_tests/test_stac.py::test_legacy_redirects[/collections/ls7_nbar_scene-/stac/collections/ls7_nbar_scene] - AssertionError: Expected /collections/ls7_nbar_scene to be redirected to:
FAILED integration_tests/test_stac.py::test_legacy_redirects[/collections/ls7_nbar_scene/items-/stac/collections/ls7_nbar_scene/items] - AssertionError: Expected /collections/ls7_nbar_scene/items to be redirected to:
FAILED integration_tests/test_stac.py::test_legacy_redirects[/collections/ls7_nbar_scene/items?datetime=2000-01-01/2000-01-01&bbox=-48.206,-14.195,-45.067,-12.272-/stac/collections/ls7_nbar_scene/items?datetime=2000-01-01%2F2000-01-01&bbox=-48.206%2C-14.195%2C-45.067%2C-12.272]
FAILED integration_tests/test_stac.py::test_legacy_redirects[/collections/ls7_nbar_scene/items/0c5b625e-5432-4911-9f7d-f6b894e27f3c-/stac/collections/ls7_nbar_scene/items/0c5b625e-5432-4911-9f7d-f6b894e27f3c] - Assert...
```

--- a/.github/workflows/lint.yml
+++ b/.github/workflows/lint.yml
@@ -27,10 +27,6 @@ jobs:
           make build
           docker-compose up -d
 
-      - name: work around permission issue
-        run: docker-compose run explorer \
-                git config --global --add safe.directory /code
-
       - name: Run linting
         run: |
           make lint-docker

--- a/.github/workflows/lint.yml
+++ b/.github/workflows/lint.yml
@@ -28,7 +28,7 @@ jobs:
           docker-compose up -d
 
       - name: work around permission issue
-        run: docker-compose exec explorer \
+        run: docker-compose run explorer \
                 git config --global --add safe.directory /code
 
       - name: Run linting

--- a/.github/workflows/lint.yml
+++ b/.github/workflows/lint.yml
@@ -18,9 +18,6 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
-      - name: work around permission issue
-        run: git config --global --add safe.directory /code
-
       - uses: actions/checkout@v3
         with:
           fetch-depth: 0
@@ -29,6 +26,10 @@ jobs:
         run: |
           make build
           docker-compose up -d
+
+      - name: work around permission issue
+        run: docker-compose exec explorer \
+                git config --global --add safe.directory /code
 
       - name: Run linting
         run: |

--- a/.github/workflows/lint.yml
+++ b/.github/workflows/lint.yml
@@ -18,6 +18,9 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
+      - name: work around permission issue
+        run: git config --global --add safe.directory /code
+
       - uses: actions/checkout@v3
         with:
           fetch-depth: 0

--- a/.github/workflows/lint.yml
+++ b/.github/workflows/lint.yml
@@ -18,7 +18,7 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
         with:
           fetch-depth: 0
 

--- a/.pre-commit-config.yaml
+++ b/.pre-commit-config.yaml
@@ -31,7 +31,7 @@ repos:
       - id: flake8-markdown
   # Lint Jinja2 templates
   - repo: https://github.com/thibaudcolas/curlylint
-    rev: "v0.13.0"
+    rev: "v0.13.1"
     hooks:
     - id: curlylint
   # Lint Yaml files

--- a/Makefile
+++ b/Makefile
@@ -32,6 +32,9 @@ format: ## Reformat all Python code
 
 .PHONY: lint
 lint: ## Run all Python linting checks
+	pwd
+	ls -la
+	git config --global --add safe.directory /code
 	git tag
 	python3 setup.py check -rms
 	pre-commit run -a

--- a/Makefile
+++ b/Makefile
@@ -32,7 +32,6 @@ format: ## Reformat all Python code
 
 .PHONY: lint
 lint: ## Run all Python linting checks
-	git tag
 	python3 setup.py check -rms
 	pre-commit run -a
 

--- a/Makefile
+++ b/Makefile
@@ -32,6 +32,7 @@ format: ## Reformat all Python code
 
 .PHONY: lint
 lint: ## Run all Python linting checks
+	git tag
 	python3 setup.py check -rms
 	pre-commit run -a
 

--- a/Makefile
+++ b/Makefile
@@ -32,10 +32,6 @@ format: ## Reformat all Python code
 
 .PHONY: lint
 lint: ## Run all Python linting checks
-	pwd
-	ls -la
-	git config --global --add safe.directory /code
-	git tag
 	python3 setup.py check -rms
 	pre-commit run -a
 

--- a/Makefile
+++ b/Makefile
@@ -32,6 +32,7 @@ format: ## Reformat all Python code
 
 .PHONY: lint
 lint: ## Run all Python linting checks
+	git config --global --add safe.directory /code
 	python3 setup.py check -rms
 	pre-commit run -a
 

--- a/cubedash/summary/_stores.py
+++ b/cubedash/summary/_stores.py
@@ -1567,7 +1567,7 @@ class SummaryStore:
         )
         self._product.cache_clear()
 
-    @lru_cache()
+    @lru_cache()  # noqa: B019
     def _get_srid_name(self, srid: int):
         """
         Convert an internal postgres srid key to a string auth code: eg: 'EPSG:1234'

--- a/cubedash/summary/_summarise.py
+++ b/cubedash/summary/_summarise.py
@@ -229,7 +229,7 @@ class Summariser:
         )
         return begin_time, end_time, where_clause
 
-    @lru_cache()
+    @lru_cache()  # noqa: B019
     def _get_srid_name(self, srid: int):
         """
         Convert an internal postgres srid key to a string auth code: eg: 'EPSG:1234'

--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -7,7 +7,7 @@ services:
       args:
         ENVIRONMENT: deployment
     ports:
-      - 80:8080
+      - 81:8080
     environment:
       # - DB_HOSTNAME=host.docker.internal
       - DB_HOSTNAME=postgres
@@ -29,7 +29,7 @@ services:
       - POSTGRES_PASSWORD=opendatacubepassword
       - POSTGRES_USER=opendatacube
     ports:
-      - 5432:5432
+      - 5431:5432
     restart: always
 
   # Needed for testing HTTPS

--- a/integration_tests/test_page_loads.py
+++ b/integration_tests/test_page_loads.py
@@ -576,7 +576,7 @@ def test_general_dataset_redirect(client: FlaskClient):
     assert rv.status_code == 302
     assert (
         rv.location
-        == "http://localhost/products/ls7_level1_scene/datasets/57848615-2421-4d25-bfef-73f57de0574d"
+        == "/products/ls7_level1_scene/datasets/57848615-2421-4d25-bfef-73f57de0574d"
     )
 
 

--- a/integration_tests/test_stac.py
+++ b/integration_tests/test_stac.py
@@ -1069,7 +1069,8 @@ def test_stac_item(stac_client: FlaskClient, populated_index: Index):
             + (
                 "?datetime=2000-01-01/2000-01-01&bbox=-48.206,-14.195,-45.067,-12.272"
                 # Flask will auto-escape parameters
-                .replace(",", "%2C").replace("/", "%2F")
+                # .replace(",", "%2C")
+                .replace("/", "%2F")
             ),
         ),
         (
@@ -1080,7 +1081,7 @@ def test_stac_item(stac_client: FlaskClient, populated_index: Index):
 )
 def test_legacy_redirects(stac_client: FlaskClient, url: str, redirect_to_url: str):
     resp: Response = stac_client.get(url, follow_redirects=False)
-    assert resp.location == stac_url(redirect_to_url), (
+    assert resp.location == redirect_to_url, (
         f"Expected {url} to be redirected to:\n"
         f"             {redirect_to_url}\n"
         f"  instead of {resp.location}"

